pkg/data/cloud/platform.go

```diff
@@ -0,0 +1,63 @@
+package platform
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+// Platform type represents specific Platform types and how they map to their respective platforms.
+type Platform struct {
+	// names holds 2 unique lowercase names of the Platform (e.g., "aws"). We use a fixed-
+	// size array so that this struct remains comparable. Any of the 2 values can be used to refer
+	// to this specific Platform via Platform.PlatformByName(), but only the first (element
+	// 0) element will be the "preferred name" returned by Platform.String()
+	names [2]string
+}
+
+var (
+	AWSClassic = Platform{
+		names: [2]string{"aws", "aws-classic"},
+	}
+	AWSHCP = Platform{
+		names: [2]string{"hostedcluster", "aws-hcp"},
+	}
+	GCPClassic = Platform{
+		names: [2]string{"gcp", "gcp-classic"},
+	}
+)
+
+// String returns the "preferred name" of the Platform
+func (plat Platform) String() string {
```

Comment:
I'm not sure if this was intentional or not, but having a `func (x X) String() string` method on a struct will override how it gets output during string interpolation. Not necessarily anything wrong with this, but worth noting.

In any event, I think a name like "PreferredName" instead of "String" is more clear about what the method is doing. What are your thoughts on renaming this?

Reply:
Also, is there any reason to _not_ export "names" and let consumers access it directly?

Reply:
Leaving the overriding String method per team discussions